pkg/controllers/user/logging/clusterlogging.go

```diff
@@ -124,18 +127,49 @@ func (c *ClusterLoggingSyncer) createOrUpdateClusterConfigMap() error {
 	return utils.UpdateConfigMap(loggingconfig.ClusterConfigPath, loggingconfig.ClusterLoggingName, "cluster", c.configmaps)
 }
 
-func (c *ClusterLoggingSyncer) updateEmbeddedEndpoint() {
-	timeout := 3 * time.Minute
-	syncInterval := 10 * time.Second
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
-	go func(ctx context.Context, cancel context.CancelFunc) {
-		for range ticker.Context(ctx, syncInterval) {
-			err := utils.UpdateEmbeddedEndpoint(c.podLister, c.serviceLister, c.clusterLoggings)
-			if err != nil {
-				logrus.Debugf("updateEmbeddedEndpoint failed, %v", err)
-			} else {
-				cancel()
-			}
+func (c *ClusterLoggingLifecycle) provision(obj *v3.ClusterLogging) (newobj *v3.ClusterLogging, err error) {
+	logger, io := utils.NewClusterLoggingLogger(obj, c.clusterLoggings, v3.LoggingConditionProvisioned)
+	defer func() {
+		if err != nil {
+			logger.Infof("provision failed %s", err.Error())
+		}
+		defer io.Close()
+	}()
+	newobj = obj
+	logger.Infof("init namespace %s", loggingconfig.LoggingNamespace)
+	if err = utils.IniteNamespace(c.namespaces); err != nil {
+		return
+	}
+	logger.Infof("init configmap %s", loggingconfig.ClusterLoggingName)
+	if err = utils.InitConfigMap(c.configmaps); err != nil {
+		return
+	}
+	//embedded
+	if utils.GetClusterTarget(obj.Spec) == "embedded" {
```

Comment:
provision func will only be called when the LoggingConditionProvisioned is not true, correct?  
If the CRD is created, fluentd/configmap will be created, and LoggingConditionProvisioned is true.
User can still make change to the logging target, like switching from splunk to embedded target. However,  Since LoggingConditionProvisioned is already true, this logic will never be called! It should be inside LoggingConditionUpdated.Do.